src/main/kotlin/lotto/domain/LottoTickets.kt

```diff
@@ -4,12 +4,8 @@ import lotto.vo.Money
 
 class LottoTickets(val lottoTickets: List<LottoTicket>) {
 
-    fun matching(winningLotto: LottoTicket): LottoRanks {
-        val lottoRanks = lottoTickets.map {
-            val matchCount = winningLotto.matching(it)
-            LottoRank.of(matchCount)
-        }
-        return LottoRanks(lottoRanks)
+    fun matching(winningLotto: WinningLotto): LottoRanks {
+        return LottoRanks(lottoTickets.map { winningLotto.matching(it) })
```

Comment:
여기서도 function reference를 사용할 수 있을 것 같아요!
```suggestion
        return LottoRanks(lottoTickets.map(winningLotto::matching))
```

이미 여러 곳에서 function reference를 많이 활용하시고 계셔서 아시겠지만 function reference를 사용하면 lambda에서 함수를 호출하는것과 달리 중간에 값이 조작되지 않음을 보장할 수 있다는 이점이 있습니다.

function reference와 lambda가 decomplie 되었을때 어떤 차이점이 있는지는 다음 글에 자세히 나와있습니다.
https://proandroiddev.com/kotlin-lambda-vs-method-reference-fdbd175f6845